### PR TITLE
david+fede+jordan/1402-Proposal creation

### LINKED ISSUE #1402: Proposal creation
Scope:
- [x] create proposal creation post function in lcdClient
- [x] create proposal creation actions in governance module
- [ ] attach proposal creation to view components
- [x] write tests for actions
- [x] write tests for lcdClient functions

### LINKED ISSUE #1582: Use time instead of block for proposals
Description:

Text proposals use `time` instead of `block`. We will need to update `PageProposal`, `LiProposal` (use id instead of block);

```
// Text Proposals
type TextProposal struct {
	ProposalID   uint64       `json:"proposal_id"`   //  ID of the proposal
	Title        string       `json:"title"`         //  Title of the proposal
	Description  string       `json:"description"`   //  Description of the proposal
	ProposalType ProposalKind `json:"proposal_type"` //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}

	Status      ProposalStatus `json:"proposal_status"` //  Status of the Proposal {Pending, Active, Passed, Rejected}
	TallyResult TallyResult    `json:"tally_result"`    //  Result of Tallys

	SubmitTime     time.Time `json:"submit_time"`      //  Time of the block where TxGovSubmitProposal was included
	DepositEndTime time.Time `json:"deposit_end_time"` // Time that the Proposal would expire if deposit amount isn't met
	TotalDeposit   sdk.Coins `json:"total_deposit"`    //  Current deposit on this proposal. Initial value is set at InitialDeposit

	VotingStartTime time.Time `json:"voting_start_time"` //  Time of the block where MinDeposit was reached. -1 if MinDeposit is not reached
	VotingEndTime   time.Time `json:"voting_end_time"`   // Time that the VotingPeriod for this proposal will end and votes will be tallied
}
```

### PR DESCRIPTION
**Pull request by NodeGuy**
attach proposal creation to view components

Closes #1402
Fixes #1582 

---

* [x] Added entries in `CHANGELOG.md` with issue # and GitHub username
* [x] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

**Comment by NodeGuy** [comment]
Blocked by https://github.com/cosmos/voyager/issues/1517, which occurs when a new proposal is created.

**Comment by jbibla** [comment]
> Can't add spaces on the title

this is a tendermint UI issue - he opened a ticket for it 😄

**Comment by NodeGuy** [comment]
> Can't add spaces on the title

I reported this here https://github.com/tendermint/ui/issues/82.

> Theres 2 back buttons on the header ¯_(ツ)_/¯

Nice catch!  Fixed.

> Prefix error returns 400 once it's submitted

This used to be caused by https://github.com/cosmos/voyager/issues/1517 but has been fixed for me.  Perhaps try running `yarn build:gaia`?

**Comment by fedekunze** [comment]
Just realized that this form is currently missing the `initital_deposit` field:

https://github.com/cosmos/cosmos-sdk/blob/bb54a0de127e45713f272217f578c0abe53a5b21/x/gov/msgs.go#L16-L22

**Comment by fedekunze** [comment]
@NodeGuy there are some refactors not related to the scope of the PR (e.g: `lcdClient.js`). Do you mind if I split the PR for ease of review? I could do it tomorrow if u like

**Comment by fedekunze** [comment]
I've split the refactors into #1567 @NodeGuy @faboweb

**Comment by fedekunze** [comment]
as per discussion with @faboweb, we are going to implement modal instead of connecting the existing page

**Comment by fedekunze** [comment]
@jbibla is going to help with styling

**Comment by fedekunze** [comment]
snapshot

e2e is failing bc the last buttons are not being displayed

File: app/src/renderer/components/common/VmToolBar.vue [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
-  tm-tool-bar
+  .tm-tool-bar
     a.back(@click="back" :disabled="user.history.length === 0" v-tooltip.bottom="'Back'")
       i.material-icons arrow_back
     slot
     a.help(@click="enableModalHelp" v-tooltip.bottom="'Help'")
       i.material-icons help_outline
-    router-link.settings(to="/preferences" v-tooltip.bottom="'Preferences'")
-      i.material-icons#settings settings
+    a#settings(@click="goToPreferences" v-tooltip.bottom="'Preferences'")
```
Comment (jbibla) [inline app/src/renderer/components/common/VmToolBar.vue]: is there a reason we're not using router-link? click event with a method on an `a` tag seems strange.

Comment (fedekunze) [reply app/src/renderer/components/common/VmToolBar.vue]: yeah maybe, it was one of my desperate attempts to fix the e2e tests lol. Feel free to push changes

Comment (fedekunze) [reply app/src/renderer/components/common/VmToolBar.vue]: I'll update it for an `href`

**Comment by fedekunze** [comment]
fixing some things that were broken due to v0.26

File: app/src/renderer/components/common/VmToolBar.vue (hunk 1 above)
Comment (fedekunze) [reply app/src/renderer/components/common/VmToolBar.vue]: reverted changes

File: test/e2e/launch.js [hunk 2]
```diff
@@ -232,6 +232,12 @@ async function writeLogs(app, location) {
 async function startApp(app, awaitingSelector = `.tm-session`) {
   console.log(`Starting app`)
   await app.start()
+  if (!app.browserWindow) {
+    console.log(`No browser window`)
+    return
```
Comment (faboweb) [inline test/e2e/launch.js]: should this crash?

Comment (fedekunze) [reply test/e2e/launch.js]: It should. Do you want me to remove those lines ?

**Pull request merged**